Firestore/Source/API/FSTUserDataConverter.h

```diff
@@ -36,13 +36,19 @@ NS_ASSUME_NONNULL_BEGIN
 - (instancetype)init NS_UNAVAILABLE;
 
 - (instancetype)initWithData:(FSTObjectValue *)data
-                   fieldMask:(nullable FSTFieldMask *)fieldMask
              fieldTransforms:(NSArray<FSTFieldTransform *> *)fieldTransforms
     NS_DESIGNATED_INITIALIZER;
 
+- (instancetype)initWithData:(FSTObjectValue *)data
+                   fieldMask:(firebase::firestore::model::FieldMask)fieldMask
+             fieldTransforms:(NSArray<FSTFieldTransform *> *)fieldTransforms
+    NS_DESIGNATED_INITIALIZER;
```

Comment:
Hmm, can you have two `NS_DESIGNATED_INITIALIZER`s?

Reply:
I believe so. I think it's supposed to be present on all non-convenience initializers. There's more info on the topic here:

https://developer.apple.com/library/content/releasenotes/ObjectiveC/ModernizationObjC/AdoptingModernObjective-C/AdoptingModernObjective-C.html#//apple_ref/doc/uid/TP40014150-CH1-SW8

In this particular case, looking at the implementation below, I believe this macro is correct here. (Usual caveats that I know nothing about obj-c apply.)

Reply:
Looks like it's indeed the case, thanks for providing the reference.